engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/encode/OutputValidator.scala

```diff
@@ -0,0 +1,32 @@
+package pl.touk.nussknacker.engine.avro.encode
+
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import org.apache.avro.Schema
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{CustomNodeError, NodeId}
+import pl.touk.nussknacker.engine.api.typed.{CanBeSubclassDeterminer, SingleSubclassRestriction, typing}
+import pl.touk.nussknacker.engine.api.typed.typing.{TypedObjectTypingResult, TypingResult}
+import pl.touk.nussknacker.engine.avro.KafkaAvroBaseTransformer.SinkValueParamName
+import pl.touk.nussknacker.engine.avro.typed.AvroSchemaTypeDefinitionExtractor
+
+object OutputValidator {
+
+  def validateOutput(value: TypingResult, schema: Schema, validationMode: ValidationMode)(implicit nodeId: NodeId): Validated[CustomNodeError, Unit] = {
+    val possibleTypes = AvroSchemaTypeDefinitionExtractor.ExtendedPossibleTypes
+    //FIXME: this still does not handle optional fields validation properly for acceptUnfilledOptional == true?
```

Comment:
1. Can you change it to less offensive `TODO`, because it doens't look like it is very "hot" problem
2. I think that it is something missing in this comment. Am I right that you mean "not handle defined optional fields type type validation"? Or something like this? Because now it looks like we not handle this validation at all...

Reply:
ok